### ISSUE TITLE
`skywire-cli visor halt` not stopping the visor

### ISSUE THREAD

**Issue by 0pcom** [skycoin/skywire#1307]
`skywire-cli visor halt` results in the visor shutting down many modules, but not all of them.

The hypervisor UI is still being served

here I am testing the skywire-systray 1.0.1 binary running at the user level on linux; with the visor running at the user level, I execute the `skywire-cli visor halt` command` in a different terminal
```
[2022-07-13T09:25:41-05:00] INFO [visor:shutdown]: Begin shutdown.
[2022-07-13T09:25:41-05:00] INFO [proc:skysocks:6340b4bf197f4a3a83c79e089ca24589]: App skysocks is Stopped
2022/07/13 09:25:42 rpc.Serve: accept:accept tcp 127.0.0.1:3435: use of closed network connection
[2022-07-13T09:25:42-05:00] INFO [visor:shutdown]: Shutdown complete. Goodbye!
```

it did not actually stop, but all the logging ceases. The process does not exit in the terminal. The visor stops respondig to rpc.

here I click in the systray to open the hypervisor, which starts a different logging
```
[2022-07-13T09:25:58-05:00] INFO []: Opening hypervisor at http://localhost:8000
[2022-07-13T09:27:11-05:00] INFO []: Opening hypervisor at http://localhost:8000
[2022-07-13T09:27:41-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:27:51-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:28:01-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:28:11-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:28:21-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:28:31-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:28:41-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:28:51-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:29:02-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:29:12-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:29:22-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:29:32-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:29:42-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:29:52-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:30:02-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:30:13-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:30:23-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:30:33-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:30:43-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:30:53-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:31:03-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:31:13-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:31:23-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:31:34-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:31:44-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:31:54-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:32:04-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:32:14-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:32:24-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:32:34-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:32:45-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:32:55-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:33:05-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:33:15-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:33:25-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
[2022-07-13T09:33:35-05:00] WARN dtm.establishTracker [dmsg_tracker_manager]: Failed to re-create dmsgtracker client. client_pk=0363bcc0cefe5d2a98cb4c2105f71c69dcc378513533a586fbcdaebd13f4faf661 error="dmsg error 100 - entry is not found in discovery"
```
The hypervisor UI opens, and to my astonishment, it's indicating that the visor is still online

![image](https://user-images.githubusercontent.com/36607567/178763172-7448e6ac-a348-426c-af7f-031e3950276d.png)

Finally, I press `ctrl`+`c` in the terminal where the visor is running

```
^C[2022-07-13T09:33:39-05:00] INFO []: Closing with received signal. signal=interrupt
[2022-07-13T09:33:39-05:00] INFO [visor:shutdown]: Begin shutdown.
[2022-07-13T09:33:39-05:00] WARN [14/14] [visor:shutdown:dmsgctrl]: Module stopped with unexpected result. elapsed=20.901µs error="dmsg error 200 - local entity closed"
[2022-07-13T09:33:39-05:00] WARN [12/14] [visor:shutdown:launcher.proc_manager]: Module stopped with unexpected result. elapsed=14.382µs error="proc manager is already closed"
[2022-07-13T09:33:39-05:00] WARN [router]: closing route_manager returned error error="dmsg error 200 - local entity closed"
[2022-07-13T09:33:39-05:00] WARN [11/14] [visor:shutdown:router.serve]: Module stopped with unexpected result. elapsed=8.05226ms error="dmsg error 200 - local entity closed"
[2022-07-13T09:33:39-05:00] WARN [7/14] [visor:shutdown:dmsg_tracker_manager]: Module stopped with unexpected result. elapsed=32.597µs error="io: read/write on closed pipe"
[2022-07-13T09:33:39-05:00] WARN [6/14] [visor:shutdown:router.serve]: Module stopped with unexpected result. elapsed=61.69µs error="close unix /tmp/dmsgpty.sock: use of closed network connection"
[2022-07-13T09:33:39-05:00] WARN [3/14] [visor:shutdown:cli.listener]: Module stopped with unexpected result. elapsed=5.478µs error="close tcp 127.0.0.1:3435: use of closed network connection"
[2022-07-13T09:33:43-05:00] ERROR [2/14] [visor:shutdown:address_resolver]: Module timed out. elapsed=4.000450737s
[2022-07-13T09:33:43-05:00] WARN [1/14] [visor:shutdown:event_broadcaster]: Module stopped with unexpected result. elapsed=30.422µs error="event subscriptions is closed"
[2022-07-13T09:33:43-05:00] INFO [visor:shutdown]: Shutdown complete. Goodbye!

```

**Comment by mrpalide** [comment]
Seems `halt` cannot close hypervisor module and not related to systray, same on non-systray.

Let me check it.

Update: shutting down hypervisor module added, but not related to that. Actually when we use `Ctrl+C`, we send exit signal, but in `halt` command we missed this exit signal, so I added `os.Exit(0)` there.

**Comment by 0pcom** [comment]
Fixed by PR #1310
